client/my-sites/plans-features-main/index.jsx

```diff
@@ -109,6 +114,14 @@ export class PlansFeaturesMain extends Component {
 		return true;
 	}
 
+	isDisplayingPlansNeededForFeature() {
+		const { selectedFeature, selectedPlan, previousRoute } = this.props;
+
+		if ( selectedFeature && selectedPlan && ! previousRoute.startsWith( '/plans/' ) ) {
```

Comment:
As an extra safety check to avoid [the issue I ran into](https://github.com/Automattic/wp-calypso/pull/41044/files#r409621782), we can check the selected feature and plan is valid:

```suggestion
		if ( selectedFeature && isValidFeatureKey( selectedFeature ) && selectedPlan && getPlan( selectedPlan ) && ! previousRoute.startsWith( '/plans/' ) ) {
```